ddtrace/tracer.py

```diff
@@ -529,6 +538,8 @@ def trace(self, name, service=None, resource=None, span_type=None):
             parent2 = tracer.trace('parent2')   # has no parent span
             parent2.finish()
         """
+        self._check_new_process()
```

Comment:
Can you explain why we want to do this check upon calling `trace` rather than `start_span`?

Reply:
Yeah I was thinking of adding a comment explaining the move.

We want to do this check in order to activate the new context before the call to `get_call_context()`. 

However, we do run into issues now for `start_span` users because of the other side-effects going on in `_check_new_process()` not being run.

We _should_ be able to move the `_check_new_process()` check into `start_span` and just do it earlier. I'll try making that change.